crates/autopilot/src/solvable_orders.rs

```diff
@@ -168,31 +172,84 @@ impl SolvableOrdersCache {
     /// the case in unit tests, then concurrent calls might overwrite each
     /// other's results.
     pub async fn update(&self, block: u64) -> Result<()> {
+        const INITIAL_ORDER_CREATION_TIMESTAMP: DateTime<Utc> = DateTime::<Utc>::MIN_UTC;
```

Comment:
After thinking more about it do we even need this sentinel value?
The cached order can already be optional and I would expect once we populate the cache there also has to be a non-zero timestamp in there so we can determine by the cache being `None` that we need to fetch ALL open orders instead of the incremental update.

Reply:
With `None` some of the e2e tests fail. It's probably worth a separate PR to fix it.

Reply:
Let's please understand at least what causes the tests to fail. I don't want us to check in code without understanding why it's actually needed.

Reply:
That requires refactoring. Some of the code depends on auction always present in the DB, which is used across many of the e2e tests.
https://github.com/cowprotocol/services/blob/fb8dad32106a2a42bf9cfb5e6a7db60755fa69f3/crates/e2e/src/setup/services.rs#L303-L317
Also:
https://github.com/cowprotocol/services/blob/db3a1b578c1caddd8ec428a98db5a8039177e2cb/crates/e2e/src/setup/services.rs#L304-L317

Reply:
Would be great to refactor since the sentinel value really doesn't seem necessary but IMO that can happen after this gets merged so we can test this a bit on staging before merging to prod on Tuesday.